From cddb620c0284c6c98ade9f218c8924ea55b51fa3 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Tue, 20 Jan 2015 20:00:47 -0500
Subject: [drm] i915: demote opregion excessive timeout WARN_ONCE to DRM_INFO_ONCE

Message-id: <1421784047-1555-1-git-send-email-rclark@redhat.com>
Patchwork-id: 103223
O-Subject: [RHEL7 drm] drm/i915: demote opregion excessive timeout WARN_ONCE to DRM_INFO_ONCE
Bugzilla: 1145627
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Upstream: since drm-intel-fixes-2014-02-06
Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1145627

commit bdde5c6a258a702bdfa7d1f4ae804a7bc405e788

Author:     Jani Nikula <jani.nikula@intel.com>
AuthorDate: Fri Jan 31 15:49:08 2014 +0200
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Tue Jan 20 14:42:39 2015 -0500

    drm/i915: demote opregion excessive timeout WARN_ONCE to DRM_INFO_ONCE

    The WARN_ONCE is a bit too verbose, make it a DRM_INFO_ONCE.

    While at it, add a #define for MAX_DSLP and make the message a bit more
    informative.

    v2: use DRM_INFO_ONCE, add MAX_DSLP, pimp the message.

    Suggested-by: Chris Wilson <chris@chris-wilson.co.uk>
    Signed-off-by: Jani Nikula <jani.nikula@intel.com>
    Reviewed-by: Chris Wilson <chris@chris-wilson.co.uk>
    Signed-off-by: Daniel Vetter <daniel.vetter@ffwll.ch>

Signed-off-by: Rob Clark <rclark@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/i915/intel_opregion.c b/drivers/gpu/drm/i915/intel_opregion.c
index 3574274..f5801a3 100644
--- a/drivers/gpu/drm/i915/intel_opregion.c
+++ b/drivers/gpu/drm/i915/intel_opregion.c
@@ -227,6 +227,8 @@ struct opregion_asle {
 #define ACPI_DIGITAL_OUTPUT (3<<8)
 #define ACPI_LVDS_OUTPUT (4<<8)
 
+#define MAX_DSLP 1500
+
 #ifdef CONFIG_ACPI
 static int swsci(struct drm_device *dev, u32 function, u32 parm, u32 *parm_out)
 {
@@ -261,10 +263,11 @@ static int swsci(struct drm_device *dev, u32 function, u32 parm, u32 *parm_out)
   /* The spec says 2ms should be the default, but it's too small
    * for some machines. */
   dslp = 50;
- } else if (dslp > 500) {
+ } else if (dslp > MAX_DSLP) {
   /* Hey bios, trust must be earned. */
-  WARN_ONCE(1, "excessive driver sleep timeout (DSPL) %u\n", dslp);
-  dslp = 500;
+  DRM_INFO_ONCE("ACPI BIOS requests an excessive sleep of %u ms, "
+         "using %u ms instead\n", dslp, MAX_DSLP);
+  dslp = MAX_DSLP;
  }
 
  /* The spec tells us to do this, but we are the only user... */
-- 
1.7.1